core/api.go

```diff
@@ -135,12 +146,13 @@ func (a *api) Start() {
 
 // Stop stops the http api
 func (a *api) Stop() error {
-	ctx, cancel := context.WithCancel(context.Background())
+	// Use timeout to force a deadline
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
```

Comment:
useful for channels that take too long to close

Reply:
perf